libdrgn/dwarf_info.c

```diff
@@ -6786,6 +6786,40 @@ drgn_type_from_dwarf_internal(struct drgn_debug_info *dbinfo,
 	return NULL;
 }
 
+static struct drgn_error *
+find_enclosing_namespace(struct drgn_namespace_dwarf_index *global_namespace,
+			 struct drgn_namespace_dwarf_index *starting_namespace,
```

Comment:
Do you have plans to add cases for where `global_namespace` and `starting_namespace` are different? If not, I think let's just pass the global namespace and assume that they're the same.

Reply:
Not at the moment, was just planning on making this extensible in case we add the ability to search relative to a namespace (like via the scopes API you outlined), but sure we can just cross that bridge when we get there.